modules/core/src/main/scala/org/scalasteward/core/buildsystem/maven/MavenAlg.scala

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2018-2019 Scala Steward contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.scalasteward.core.buildsystem.maven
+
+import atto.Atto._
+import atto._
+import better.files.File
+import cats.Monad
+import cats.implicits._
+import io.chrisdavenport.log4cats.Logger
+import org.scalasteward.core.application.Config
+import org.scalasteward.core.buildsystem.BuildSystemAlg
+import org.scalasteward.core.data._
+import org.scalasteward.core.io.{FileAlg, ProcessAlg, WorkspaceAlg}
+import org.scalasteward.core.scalafix.Migration
+import org.scalasteward.core.util.Nel
+import org.scalasteward.core.vcs.data.Repo
+
+object MavenAlg {
+
+  def create[F[_]](
+      implicit
+      config: Config,
+      fileAlg: FileAlg[F],
+      processAlg: ProcessAlg[F],
+      workspaceAlg: WorkspaceAlg[F],
+      logger: Logger[F],
+      F: Monad[F]
+  ): BuildSystemAlg[F] = new BuildSystemAlg[F] {
+
+    import MavenParser._
+
+    override def containsBuild(repo: Repo): F[Boolean] = ???
+
+    override def getDependencies(repo: Repo): F[List[Scope.Dependencies]] =
+      for {
+        repoDir <- workspaceAlg.repoDir(repo)
+        listDependenciesCommand = mvnCmd(command.Clean, command.mvnDepList)
```

Comment:
@stanislav-chetvertkov I'm wondering why `clean` is called here? Is it required by `mvnDepList`?

Reply:
I guess it was an extra precaution, probably could be removed